packages/syft/src/syft/service/action/action_object.py

```diff
@@ -511,8 +513,21 @@ def reload_cache(self):
                         blob_retrieval_object,
                     )
                     return blob_retrieval_object
-                self.syft_action_data_cache = blob_retrieval_object.read()
-                self.syft_action_data_type = type(self.syft_action_data)
+
+                if isinstance(blob_retrieval_object, SyftError):
+                    raise SyftException(
+                        message=f"Failed to retrieve object from blob storage: {blob_retrieval_object.message}"
+                    )
+
+                elif isinstance(
+                    blob_retrieval_object, (BlobRetrievalByURL, BlobRetrieval)
```

Comment:
nitpick: I think we could just check for instance(blob_retrieval_object, BlobRetrieval), since BlobRetrievalByURL is a subclass of BlobRetrieval. Can we also add a comment here that this change has been done temporarily to handle gateways.

Reply:
Done [87c2676](https://github.com/OpenMined/PySyft/pull/8290/commits/87c2676d16584d461543499a0debbaf2ea51ff70)